presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -1104,6 +1104,13 @@ public void validateScan(Session session, TableHandle table)
                     List<ConnectorExpression> newProjections = ImmutableList.copyOf(result.getProjections());
                     List<ProjectionApplicationResult.Assignment> newAssignments = ImmutableList.copyOf(result.getAssignments());
 
+                    verify(
```

Comment:
Throw a PrestoException (we'll need a new error code similar to `FUNCTION_IMPLEMENTATION_ERROR`)